### ISSUE TITLE
[GENERAL SUPPORT]: generator_kwargs formatting

### ISSUE THREAD

**Issue by VMLC-PV** [facebook/Ax#5230]
### Question

In previous versions (up to 1.2.4), I used to use the following `model_kwargs` (now `generator_kwargs`) in my BOTORCH_MODULAR steps:

```python
{"torch_device":torch.device("cuda" if torch.cuda.is_available() else "cpu"),'botorch_acqf_class':qLogExpectedHypervolumeImprovement,'transforms':[RemoveFixed, Log, UnitX, StandardizeY],'surrogate_spec':SurrogateSpec(model_configs=[ModelConfig(botorch_model_class=SingleTaskGP,covar_module_class=ScaleKernel, covar_module_options={'base_kernel':MaternKernel(nu=2.5, ard_num_dims=num_free_params,)})])}]
```

But now this seems to fail and I get the following error:

```python
```File ~/miniconda3/envs/bam/lib/python3.14/site-packages/ax/generators/torch/botorch_modular/surrogate.py:572, in Surrogate._construct_model(self, dataset, search_space_digest, model_config, state_dict, refit)
    570     model.load_state_dict(state_dict)
    571 if state_dict is None or refit:
--> [572]( ~/miniconda3/envs/bam/lib/python3.14/site-packages/ax/generators/torch/botorch_modular/surrogate.py:572)     fit_botorch_model(
    573         model,  # Intentionally not using named args for dispatcher.
    574         mll_class=model_config.mll_class,
    575         mll_options=model_config.mll_options,
    576     )
    578 return model

File ~/miniconda3/envs/bam/lib/python3.14/site-packages/botorch/utils/dispatcher.py:94, in Dispatcher.__call__(self, *args, **kwargs)
     92 func = self.__getitem__(types=types)
     93 try:
---> [94]( ~/miniconda3/envs/bam/lib/python3.14/site-packages/botorch/utils/dispatcher.py:94)     return func(*args, **kwargs)
     95 except MDNotImplementedError:
     96     # Traverses registered methods in order, yields whenever a match is found
     97     funcs = self.dispatch_iter(*types)

File ~/miniconda3/envs/bam/lib/python3.14/site-packages/ax/generators/torch/botorch_modular/utils.py:823, in _fit_botorch_model_gpytorch(model, mll_class, mll_options)
    821 mll_options = mll_options or {}
    822 mll = mll_class(likelihood=model.likelihood, model=model, **mll_options)
--> [823]( ~/miniconda3/envs/bam/lib/python3.14/site-packages/ax/generators/torch/botorch_modular/utils.py:823) fit_gpytorch_mll(mll)

File ~/miniconda3/envs/bam/lib/python3.14/site-packages/botorch/fit.py:141, in fit_gpytorch_mll(mll, closure, optimizer, closure_kwargs, optimizer_kwargs, **kwargs)
    132 if isinstance(mll, _ApproximateMarginalLogLikelihood):
    133     return _fit_fallback_approximate(
    134         mll=mll,
    135         closure=closure,
   (...)    138         **kwargs,
    139     )
--> [141]( ~/miniconda3/envs/bam/lib/python3.14/site-packages/botorch/fit.py:141) return _fit_fallback(
    142     mll=mll,
    143     closure=closure,
    144     closure_kwargs=closure_kwargs,
    145     optimizer_kwargs=optimizer_kwargs,
    146     **kwargs,
    147 )

File ~/miniconda3/envs/bam/lib/python3.14/site-packages/botorch/fit.py:241, in _fit_fallback(mll, closure, optimizer, closure_kwargs, optimizer_kwargs, max_attempts, pick_best_of_all_attempts, warning_handler, caught_exception_types, **ignore)
    239 with catch_warnings(record=True) as warning_list:
    240     simplefilter("always", category=OptimizationWarning)
--> [241]( ~/miniconda3/envs/bam/lib/python3.14/site-packages/botorch/fit.py:241)     result = optimizer(mll, closure=closure, **optimizer_kwargs)
    243 # Resolve warnings and determine whether or not to retry
    244 success = True

File ~/miniconda3/envs/bam/lib/python3.14/site-packages/botorch/optim/fit.py:103, in fit_gpytorch_mll_scipy(mll, parameters, bounds, closure, closure_kwargs, method, options, callback, timeout_sec)
     97 model = mll.model
     98 if (
     99     closure is None
    100     and isinstance(model, BatchedMultiOutputGPyTorchModel)
    101     and model._aug_batch_shape.numel() > 1
    102 ):
--> [103]( ~/miniconda3/envs/bam/lib/python3.14/site-packages/botorch/optim/fit.py:103)     return _fit_gpytorch_mll_scipy_independent(
    104         mll=mll,
    105         parameters=parameters,
    106         bounds=bounds,
    107         options=options,
    108         callback=callback,
    109         timeout_sec=timeout_sec,
    110     )
    112 # Resolve ``parameters`` and update default bounds
    113 _parameters, _bounds = get_parameters_and_bounds(mll)

File ~/miniconda3/envs/bam/lib/python3.14/site-packages/botorch/optim/fit.py:278, in _fit_gpytorch_mll_scipy_independent(mll, parameters, bounds, options, callback, timeout_sec)
    275 bounds_np = get_per_element_bounds(parameters, bounds, batch_shape)
    277 # Get initial state
--> [278]( ~/miniconda3/envs/bam/lib/python3.14/site-packages/botorch/optim/fit.py:278) x0 = batched_closure.state  # (batch_size, per_element_size)
    280 # Resolve options for fmin_l_bfgs_b_batched
    281 _recognized_options = {
    282     "gtol",
    283     "maxiter",
   (...)    288     "factr",
    289 }

File ~/miniconda3/envs/bam/lib/python3.14/site-packages/botorch/optim/closures/core.py:193, in BatchedNDarrayOptimizationClosure.state(self)
    190 for name, param in self.parameters.items():
    191     size = self._trailing_sizes[name]
    192     out[:, index : index + size] = as_ndarray(
--> [193]( ~/miniconda3/envs/bam/lib/python3.14/site-packages/botorch/optim/closures/core.py:193)         param.detach().reshape(self.batch_size, size), dtype=np_float64
    194     )
    195     index += size
    196 return out

```
This seems to happen as soon as I specify a `covar_module_class` whether it is `ScaleKernel` or `MaternKernel`
The following runs just fine:
```python
{"torch_device":torch.device("cuda" if torch.cuda.is_available() else "cpu"),'botorch_acqf_class':qLogExpectedHypervolumeImprovement,'transforms':[RemoveFixed, Log, UnitX, StandardizeY],'surrogate_spec':SurrogateSpec(model_configs=[ModelConfig(botorch_model_class=SingleTaskGP,)])}
```
I don't really get how to fix that...

### Please provide any relevant code snippet if applicable.

```shell

```

### Code of Conduct

- [x] I agree to follow this Ax's Code of Conduct

**Comment by VMLC-PV** [comment]
Nevermind I may I found a fix:
```Python
kernel = ScaleKernel(MaternKernel(nu=2.5, ard_num_dims=num_free_params))
Model = SingleTaskGP
Model.cont_kernel_factory = kernel
```

and then:
```Python
{"torch_device":torch.device("cuda" if torch.cuda.is_available() else "cpu"),'botorch_acqf_class':qLogExpectedHypervolumeImprovement,'transforms':[RemoveFixed, Log, UnitX, StandardizeY],'surrogate_spec':SurrogateSpec(model_configs=[ModelConfig(botorch_model_class=Model,) ])}
```

**Comment by saitcakmak** [comment]
Hi @VMLC-PV. I don't think this workaround uses the custom kernel:
```
kernel = ScaleKernel(MaternKernel(nu=2.5, ard_num_dims=num_free_params))
Model = SingleTaskGP
Model.cont_kernel_factory = kernel
```
`SingleTaskGP` does not have or use a `cont_kernel_factory` argument. My guess is this ends up using `SingleTaskGP` with its default kernel.

The issue seems to be due to the custom kernel not being compatible with the multi-output model this ends up fitting. You can either add a batch shape argument to the kernel (equal to num_outputs), or pass in `allow_batched_models=False` to have it fit independent models for each output.

**Comment by saitcakmak** [comment]
There is also a `ScaleMaternKernel` wrapper class in Ax that may make it slightly simpler to specify the kernel:
https://github.com/facebook/Ax/blob/main/ax/generators/torch/botorch_modular/kernels.py#L35
It has a custom input constructor implemented that can also extract `ard_num_dims` from the transformed search space for you: https://github.com/facebook/Ax/blob/main/ax/generators/torch/botorch_modular/input_constructors/covar_modules.py#L68

**Comment by VMLC-PV** [comment]
Thank you so much for catching this!
As suggested, I now use the `ScaleMaternKernel`  wrapper class.
